.github/workflows/multi_arch_ci.yml

```diff
@@ -0,0 +1,151 @@
+  # Setup job - compute configuration
+  # --------------------------------------------------------------------------
+  setup:
+    name: Setup
+    runs-on: ubuntu-24.04
+    outputs:
+      amdgpu_families_json: ${{ steps.configure.outputs.amdgpu_families_json }}
+      enabled_stages: ${{ steps.configure.outputs.enabled_stages }}
+      artifact_group: ${{ steps.configure.outputs.artifact_group }}
+      rocm_package_version: ${{ steps.version.outputs.rocm_package_version }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5.0.1
+
+      - name: Configure pipeline
+        id: configure
+        run: |
+          # Parse families into JSON array
+          IFS=',' read -ra FAMILIES <<< "${{ inputs.linux_amdgpu_families }}"
+          JSON_ARRAY=$(printf '%s\n' "${FAMILIES[@]}" | jq -R . | jq -s -c .)
+          echo "amdgpu_families_json=${JSON_ARRAY}" >> $GITHUB_OUTPUT
+          echo "Families JSON: ${JSON_ARRAY}"
+
+          # Determine enabled stages
+          STAGES="${{ inputs.stages }}"
+          if [ -z "$STAGES" ]; then
+            STAGES="foundation,compiler-runtime,math-libs,comm-libs,dctools-core,profiler-apps"
+          fi
+          echo "enabled_stages=${STAGES}" >> $GITHUB_OUTPUT
+          echo "Enabled stages: ${STAGES}"
+
+          # Create unified artifact group name (all families share one run)
+          echo "artifact_group=multi-arch-${{ inputs.build_variant }}" >> $GITHUB_OUTPUT
+
+      - name: Compute package version
+        id: version
+        run: |
+          if [ "${{ inputs.package_version }}" = "ADHOCBUILD" ]; then
+            VERSION=$(python ./build_tools/compute_rocm_package_version.py --release-type=dev | grep -oP 'rocm_package_version=\K.*')
+            echo "rocm_package_version=${VERSION}" >> $GITHUB_OUTPUT
+          else
+            echo "rocm_package_version=${{ inputs.package_version }}" >> $GITHUB_OUTPUT
+          fi
+
+  # --------------------------------------------------------------------------
+  # Linux Build - Sharded Pipeline
+  # --------------------------------------------------------------------------
+  linux_build:
+    name: Linux Build (${{ inputs.build_variant }})
+    needs: setup
+    uses: ./.github/workflows/multi_arch_build_portable_linux.yml
+    secrets: inherit
+    with:
+      amdgpu_families_json: ${{ needs.setup.outputs.amdgpu_families_json }}
+      artifact_group: ${{ needs.setup.outputs.artifact_group }}
+      build_variant: ${{ inputs.build_variant }}
+      package_version: ${{ needs.setup.outputs.rocm_package_version }}
+      enabled_stages: ${{ needs.setup.outputs.enabled_stages }}
+    permissions:
+      contents: read
+      id-token: write
+
+  # --------------------------------------------------------------------------
+  # Summary
+  # --------------------------------------------------------------------------
+  ci_summary:
+    name: CI Summary
```

Comment:
Choose a unique name (`ci_summary` is already taken):
```suggestion
  multi_arch_ci_summary:
    name: Multi-Arch CI Summary
```

The job name is used for status checks and we need to be able to easily disambiguate them: 

<img width="802" height="525" alt="Image" src="https://github.com/user-attachments/assets/4586dde6-c752-4a2a-81ea-172727224a8e" />